File: LibrarySpring/src/main/java/com/upce/libraryspring/user/UserService.java
package com.upce.libraryspring.user;

import com.upce.libraryspring.user.dto.UserDto;

import java.util.List;

public interface UserService {
    List<UserDto> getUsers();
    UserDto getUserByUsername(String username);
    UserDto updateUserByUsername(String username, UserDto userDto);
    void deleteByUsername(String username);
}

File: LibraryReact/library-react/src/services/UserService.js
import * as AxiosAdapter from "../adapters/AxiosAdapter";

class UserService{
    getUserByUsername(username){
        return AxiosAdapter.getReq(`/users/${username}`);
    }

    updateUserByUsername(username, userDto){
        return AxiosAdapter.putReq(`/users/${username}`, userDto);
    }
}